src/geom/Endpoint.cc

```diff
@@ -142,16 +159,19 @@ std::shared_ptr<ast::Frame> SpherePointEndpoint::makeFrame() const {
 }
 
 void SpherePointEndpoint::normalizeFrame(std::shared_ptr<ast::Frame> framePtr) const {
-    auto skyFramePtr = std::dynamic_pointer_cast<ast::SkyFrame>(framePtr);
+    // call getFrame in case framePtr points to a FrameSet
+    auto skyFramePtr = std::dynamic_pointer_cast<ast::SkyFrame>(getFrame(framePtr));
     if (!skyFramePtr) {
         std::ostringstream os;
-        os << "frame is a \"" << framePtr->getClass() << "\", not a SkyFrame";
+        os << "frame is a " << getFrame(framePtr)->getClass() << ", not a SkyFrame";
```

Comment:
Thanks for catching this. I'll fix `PointEndpoint<N>::normalizeFrame` to show the frame class (not `FrameSet`)